src/imars3d/backend/dataio/data.py

```diff
@@ -357,26 +361,48 @@ def _get_filelist_by_dir(
         embedded in the ct file to find obs with similar metadata.
     """
     # sanity check
-    # -- radiograph
+    ##########
+    # -- Process input argument ct_dir for radiographs
     if not Path(ct_dir).exists():
         logger.error(f"ct_dir {ct_dir} does not exist.")
         raise ValueError("ct_dir does not exist.")
     else:
         ct_dir = Path(ct_dir)
-    # -- open beam
-    if not Path(ob_dir).exists():
-        logger.error(f"ob_dir {ob_dir} does not exist.")
-        raise ValueError("ob_dir does not exist.")
+    ##########
+    # -- Process input argument ob_dir for open beam directories
+    if isinstance(ob_dir, (str, Path)):  # single directory
+        open_beam_dirs = [ob_dir]  # cast the single directory to a list of input directories
+    elif isinstance(ob_dir, (list, tuple)):  # multiple input directories, assumed items are of FlexPath type
+        open_beam_dirs = ob_dir
     else:
-        ob_dir = Path(ob_dir)
-    # -- dark current
+        raise ValueError("ob_dir must be either a string or a list of strings")
+    # convert each directory string to a pathlib.Path object, and validate for existence
+    for i, data_dir in enumerate(open_beam_dirs):
+        if not Path(data_dir).exists():
+            logger.error(f"open beam directory {data_dir} does not exist.")
+            raise ValueError(f"open beam directory {data_dir} does not exist.")
+        else:
```

Comment:
The `else` doesn't need to exist if everything is converted in L374-376

Reply:
Conversion doesn't check for existence but the `raise` does makes the `else:` a moot point. I'll get rid of it